core/lib/eth_client/src/types.rs

```diff
@@ -129,6 +134,64 @@ pub struct SignedCallResult {
     pub hash: H256,
 }
 
+impl SignedCallResult {
+    pub fn new(
+        raw_tx: RawTransactionBytes,
+        max_priority_fee_per_gas: U256,
+        max_fee_per_gas: U256,
+        nonce: U256,
+        hash: H256,
+    ) -> Self {
+        Self {
+            raw_tx,
+            max_priority_fee_per_gas,
+            max_fee_per_gas,
+            nonce,
+            hash,
+        }
+    }
+
+    pub fn raw_tx(&self, blob_tx_sidecar: Option<&EthTxBlobSidecar>) -> RawTransactionBytes {
```

Comment:
can you explain what this method does? Also I'm a little bit surprised, that we are adding this 'blob tx sidecar' from the side (as it seems to really modify the interface).
Maybe have 2 methods instead? (raw_tx) & raw_tx_with_blob ? 

(the second one could also check - if we can - that the transaction is really a 'blob type' transaction? )

Reply:
Before this patch the `SignedCallResult` contained a public field `raw_tx` which is essentially a transaction encoded into a byte form. We only supported 1559 transactions and so it made sense to be as it is. Important thing: for 1559 the signed transaction blob prefixed with a byte `0x02` to indicate the tx type is the network form of the transaction, end of story. With the instroduction of 4844 transactions that changes; the signed transaction blob (transaction + signature) + a byte prefix for type is no longer the complete network form for them, there is a new thing ("sidecar") that contains blobs, proofs and commitments + a new way it is all encoded together with the transaction (params + signature).

We want an interface that would leave no space for errors like: you wanted a 4844 tx, but actually went ahead and sent `raw_tx` field which in itself is not enough for it. All possible solutions to this are not ideal: you may misuse the current interface; you may misuse the interface proposed by you; currently we have no typesafe way of telling one tx from another.

Reply:
Should blob sidecar be a part of the `new` and raw_tx will return the correct value depends on it?

Reply:
the only place where we call this `new` method is in `sign_prepared_tx_for_addr` and so we would need to pass the optional sidecar also there although it would be never used there. Which is probably even less ergonomic